tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetInclusionRuleWithArrays.scala

```diff
@@ -45,37 +47,50 @@ class SetInclusionRuleWithArrays(rewriter: SymbStateRewriter) extends SetInclusi
 
   // TODO: similar to SetInRuleWithArrays.powSetIn, potential for refactoring.
   private def subset(state: SymbState, leftCell: ArenaCell, rightCell: ArenaCell, rightIsPowSet: Boolean): SymbState = {
-    // We check if the elements of leftCell are in rightCell, in order to establish the subset relation.
-    // If rightCell is a power set, we check if the elements in the sets of leftCell are in the domain of rightCell.
-    val leftElems =
-      if (rightIsPowSet) state.arena.getHas(leftCell).flatMap(state.arena.getHas)
-      else state.arena.getHas(leftCell)
+    val leftElems = state.arena.getHas(leftCell)
+    val leftElemsElems = leftElems.flatMap(state.arena.getHas)
```

Comment:
There is a distinction, but it is now in the conditional below. I had to rework this method because adding the factory calls exposed a bug. Before, if I remember correctly, we were checking, for `{{1,2},{3,4}} \subseteq SUBSET {1,2,3,4}`, if
```
1 \in {1,2,3,4} /\ 1 \in {{1,2},{3,4}}
2 \in {1,2,3,4} /\ 2 \in {{1,2},{3,4}}
3 \in {1,2,3,4} /\ 3 \in {{1,2},{3,4}}
4 \in {1,2,3,4} /\ 4 \in {{1,2},{3,4}}
```
which is obviously wrong. I though at the time that using `flatMap` would make constraints of the form `1 \in {1,2}`, but that was not happening. Now we recursively check the subset relation, i.e.,
```
{1,2} \subseteq {1,2,3,4}
{3,4} \subseteq {1,2,3,4}
```